wodles/aws/aws-s3.py

```diff
@@ -714,6 +776,112 @@ def iter_regions_and_accounts(self, account_id, regions):
+    :param profile: AWS profile
+    :param iam_role_arn: IAM Role
+    :param only_logs_after: Date after which obtain logs.
+    :param region: Region of service
+    """
+    def __init__(self, access_key, secret_key, aws_profile, iam_role_arn,
+        only_logs_after, region='us-east-1'):
+        # SQL queries
+        self.sql_create_table = """
+                            CREATE TABLE
+                                inspector (
+                                    scan_date 'text' NOT NULL,
+                                    PRIMARY KEY (scan_date));"""
+
+        self.sql_insert_value = """
+                                INSERT INTO
+                                    inspector (scan_date)
+                                VALUES
+                                    ('{}');"""
+
+        self.sql_find_last_scan = """
+                                SELECT
+                                    scan_date
+                                FROM
+                                    inspector
+                                ORDER BY
+                                    scan_date DESC
+                                LIMIT 1;"""
+        # DB and table names
+        self.db_name = 's3_inspector'
+        self.db_table_name = 'inspector'
+
+        AWSService.__init__(self, access_key=access_key, secret_key=secret_key,
+            aws_profile=aws_profile, iam_role_arn=iam_role_arn, only_logs_after=only_logs_after,
+            service_name='inspector', region=region)
+
+    def send_describe_findings(self, arn_list):
+        if len(arn_list) == 0:
+            debug('+++ There are not new events', 1)
+        else:
+            debug('+++ Processing new events...', 1)
+            response = self.client.describe_findings(findingArns=arn_list)['findings']
+            for elem in response:
+                self.send_msg(self.format_message(elem))
+
+    def get_alerts(self):
+        initial_date = self.get_last_log_date()
+        self.init_db(self.sql_create_table)
+        try:
+            # if DB is empty write initial date
+            self.db_cursor.execute(self.sql_insert_value.format(initial_date))
+            self.db_cursor.execute(self.sql_find_last_scan)
+            last_scan = self.db_cursor.fetchone()[0]
+        except sqlite3.IntegrityError:
+            self.db_cursor.execute(self.sql_find_last_scan)
+            last_scan = self.db_cursor.fetchone()[0]
+        datetime_last_scan = datetime.strptime(last_scan, '%Y-%m-%d %H:%M:%S.%f')
+        # get current time (UTC)
+        datetime_current = datetime.utcnow()
+        # describe_findings only retrieves 100 results per call
+        response = self.client.list_findings(maxResults=100, filter={'creationTimeRange':
+            {'beginDate': datetime_last_scan, 'endDate': datetime_current}})
+        self.send_describe_findings(response['findingArns'])
+        # iterate if there are more elements
+        while 'nextToken' in response:
```

Comment:
You're doing the same call to `list_findings` twice to iterate over results, I'd do the following:
```suggestion
        for response in itertools.takewhile(lambda x: x['nextToken'], self.client.list_findings(maxResults=100, filter{'creationTimeRange': {'beginDate': datetime_last_scan, 'endDate': datetime_current}}):
```

Reply:
Done!